network/tlskc.go

```diff
@@ -0,0 +1,99 @@
+package network
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
+	"math/big"
+	"time"
+
+	"net"
+
+	"github.com/dedis/cothority/log"
+)
+
+type TLSCertPEM []byte
+type TLSKeyPEM []byte
+
+// NewTLSCert returns a x509-certificate valid for all CommonNames.
+func NewTLSCert(serial *big.Int, country, org, orgUnit string,
+	validYear int, subjectKeyID []byte,
+	ips []net.IP) *x509.Certificate {
+	return &x509.Certificate{
+		SerialNumber: serial,
+		Subject: pkix.Name{
+			Country:            []string{country},
+			Organization:       []string{org},
+			OrganizationalUnit: []string{orgUnit},
+			CommonName:         "*",
```

Comment:
Good question - we'll be connecting using IP-addresses anyway, except with the simulations, where it uses 'localhost'. We can remove the `CommonName` and add a `DNSNames`. Or set that later?